network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -252,4 +264,20 @@ public String toString() {
         .toString();
     }
   }
+
+  private void saveRegisteredExecutors() throws IOException {
+    if (registeredExecutorFile != null) {
+      logger.info("Saving registered executors to {}", registeredExecutorFile);
+      ObjectOutputStream out = new ObjectOutputStream(new FileOutputStream(registeredExecutorFile));
+      // synchronize so we can write out the size :(
```

Comment:
This is already called form a synchronized block in `registerExecutor`.

Also, any reason not to just serialize the map? Aren't `ConcurrentHashMap`s serializable?